### PR TITLE
aaron-lane-delint-user-data

### PR DESCRIPTION
**Pull request by aaron-lane**
## Background

Linting is included as part of the new CI system. As the module redesign was done before we decided on the linting rules to enforce, there are a number of existing issues that need to be addressed. This branch focuses on the user data module.

[Asana task](https://app.asana.com/0/1181500399442529/1200170182808309/f)

## How Has This Been Tested

I'm running tflint locally as I work, and our GitHub Actions configuration will confirm that the issues are fixed.

### CODE REVIEW

File: modules/user_data/variables.tf [hunk 1]
```diff
@@ -1,23 +1,51 @@
...
-  default = ""
+  default     = ""
+  description = "The password which is required to create connections with the Redis Elasticache replication group."
+  type        = string
 }
 variable "redis_port" {
-  default = ""
+  default     = ""
+  description = "The port number on which the Redis Elasticache replication group accepts connections."
+  type        = string
```
Comment (anniehedgpeth) [inline modules/user_data/variables.tf:48]: Just making sure that the empty string defaults are on purpose.

**Review by anniehedgpeth — APPROVED** [review]

File: modules/user_data/variables.tf (hunk 1 above)
Comment (aaron-lane) [reply modules/user_data/variables.tf:48]: I think we should consider updating them to be `null`, but that may require other changes beyond the scope of this pull request.

Comment (rogeruiz) [reply modules/user_data/variables.tf:48]: I don't think this one should be `null` cause it gets concatenated with the host and passed as a string to the config.

**Pull request merged**